### PR TITLE
Fix display duplicate vtable in multiple use “aaa” command

### PR DESCRIPTION
**Pull request by Basstorm**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

When using `aaa` command to analyze vtables, the `rz_analysis_class_vtable_set` function in `class.c` did not check if vtable has been added to the context. Which leads to a result that when using `aaa` command many times, the output has many duplicate vtables.
```
[0x000010e0]> aaa
[x] Analyze all flags starting with sym. and entry0 (aa)
[x] Analyze function calls (aac)
[x] Analyze len bytes of instructions for references (aar)
[x] Check for vtables
[x] Type matching analysis for all functions (aaft)
[x] Propagate noreturn information
[x] Use -AA or aaaa to perform additional experimental analysis.
[0x000010e0]> acll
Polygon
  (vtable at 0x3d38)
  virtual_0 @ 0x13d4 (vtable + 0x0)
  virtual_8 @ 0x13e8 (vtable + 0x8)
Rectangle: Polygon
  (vtable at 0x3d18)
  virtual_0 @ 0x13fc (vtable + 0x0)
  virtual_8 @ 0x141c (vtable + 0x8)
Triangle: Polygon
  (vtable at 0x3cf8)
  virtual_0 @ 0x1430 (vtable + 0x0)
  virtual_8 @ 0x1458 (vtable + 0x8)
[0x000010e0]> aaa
[x] Analyze all flags starting with sym. and entry0 (aa)
[x] Analyze function calls (aac)
[x] Analyze len bytes of instructions for references (aar)
[x] Check for vtables
[x] Type matching analysis for all functions (aaft)
[x] Propagate noreturn information
[x] Use -AA or aaaa to perform additional experimental analysis.
[0x000010e0]> acll
Polygon
  (vtable at 0x3d38)
  (vtable at 0x3d38)
  virtual_0 @ 0x13d4 (vtable + 0x0)
  virtual_8 @ 0x13e8 (vtable + 0x8)
Rectangle: Polygon
  (vtable at 0x3d18)
  (vtable at 0x3d18)
  virtual_0 @ 0x13fc (vtable + 0x0)
  virtual_8 @ 0x141c (vtable + 0x8)
Triangle: Polygon
  (vtable at 0x3cf8)
  (vtable at 0x3cf8)
  virtual_0 @ 0x1430 (vtable + 0x0)
  virtual_8 @ 0x1458 (vtable + 0x8)
[0x000010e0]> aaa
[x] Analyze all flags starting with sym. and entry0 (aa)
[x] Analyze function calls (aac)
[x] Analyze len bytes of instructions for references (aar)
[x] Check for vtables
[x] Type matching analysis for all functions (aaft)
[x] Propagate noreturn information
[x] Use -AA or aaaa to perform additional experimental analysis.
[0x000010e0]> acll
Polygon
  (vtable at 0x3d38)
  (vtable at 0x3d38)
  (vtable at 0x3d38)
  virtual_0 @ 0x13d4 (vtable + 0x0)
  virtual_8 @ 0x13e8 (vtable + 0x8)
Rectangle: Polygon
  (vtable at 0x3d18)
  (vtable at 0x3d18)
  (vtable at 0x3d18)
  virtual_0 @ 0x13fc (vtable + 0x0)
  virtual_8 @ 0x141c (vtable + 0x8)
Triangle: Polygon
  (vtable at 0x3cf8)
  (vtable at 0x3cf8)
  (vtable at 0x3cf8)
  virtual_0 @ 0x1430 (vtable + 0x0)
  virtual_8 @ 0x1458 (vtable + 0x8)

```

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: librz/analysis/class.c [hunk 1]
```diff
@@ -878,10 +878,23 @@ RZ_API RzVector /*<RzAnalysisVTable>*/ *rz_analysis_class_vtable_get_all(RzAnaly
 }
 
 RZ_API RzAnalysisClassErr rz_analysis_class_vtable_set(RzAnalysis *analysis, const char *class_name, RzAnalysisVTable *vtable) {
+	/* Check if vtable exists before setting it */
+	RzVector /*<RzAnalysisVTable>*/ *vtables = rz_analysis_class_vtable_get_all(analysis, class_name);
```
Comment (ret2libc) [inline librz/analysis/class.c:882]: you should free this vector because it is dynamically allocated.

**Review by ret2libc — CHANGES_REQUESTED** [review]

File: librz/analysis/class.c (hunk 1 above)
Comment (Basstorm) [reply librz/analysis/class.c:882]: sorry for my cheap mistake..

**Comment by XVilka** [comment]
Could you please add a test for this case? It would be very helpful for the ongoing RzTypes refactoring.

**Comment by Basstorm** [comment]
Test cases have been added.

**Review by XVilka — APPROVED** [review]

File: librz/analysis/class.c (hunk 1 above)
Comment (ret2libc) [reply librz/analysis/class.c:882]: No problem :)

**Pull request merged**